eval/vals/concat.go

```diff
@@ -0,0 +1,51 @@
+package vals
+
+import (
+	"errors"
+	"fmt"
+)
+
+type (
```

Comment:
I would rather not use a `type` block, and have those types documented. But I'm not delaying this PR futher.